src/mococrw/hash.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "mococrw/openssl_wrap.h"
+
+namespace mococrw
+{
+
+std::vector<uint8_t> sha256(const std::vector<uint8_t> &message);
+std::vector<uint8_t> sha256(const uint8_t* message, size_t messageLength);
+
+std::vector<uint8_t> sha512(const std::vector<uint8_t> &message);
+std::vector<uint8_t> sha512(const uint8_t* message, size_t messageLength);
+
+class Hash
+{
+public:
+    static Hash sha256();
+    static Hash sha512();
+    std::vector<uint8_t> digest();
+    std::string hexdigest();
+    size_t getDigestLengthInBytes() const;
```

Comment:
getDigestLengthInBytes() does not fit into the concept of Hash implemented as a builder neither a stand-alone function. In case of a stand-alone function, you can't access it at all. In case of a builder, you would have to break the chain for update().update().digest() calls.

Maybe a static method or a static variable?

Reply:
agree, because vector is anyway the only type we return and it has the length